pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1260,6 +1266,23 @@ func generateDeprovisionRequest(cd *hivev1.ClusterDeployment) *hivev1.ClusterDep
 	return req
 }
 
+func generatePullSecretObj(pullSecret string, cd *hivev1.ClusterDeployment) *corev1.Secret {
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-pull-secret", cd.Name),
```

Comment:
This will conflict with the one create-cluster makes, suggest clusterName-merged-pull-secret-RANDOM suffix, and then we store the generated name into cd.Status.PullSecret. If no merged pull secret is in play, then cd.Status.PullSecret should be a direct copy of cd.Spec.PullSecret.